### PR TITLE
Update vendor dependency backward-cpp to v1.5

### PR DESCRIPTION
**Pull request by wjwwood**
Alternative to https://github.com/osrf/osrf_testing_tools_cpp/issues/44.

See also:

- https://github.com/bombela/backward-cpp/tree/v1.5#as-a-subdirectory

### CODE REVIEW

**Comment by wjwwood** [comment]
Actually it looks like there's a problem with how the new integration works on Linux, I'm looking into the alternative method.

**Comment by wjwwood** [comment]
Ok, after testing I switched to a modified version of this suggestion: https://github.com/bombela/backward-cpp/tree/v1.5#modifying-cmake_module_path

CI (full CI because I don't know what is using this exactly):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10301/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5834/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8386/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10202/)

**Comment by wjwwood** [comment]
There are some CMake warnings on Windows I have to look into.

**Comment by wjwwood** [comment]
Full CI (Fast-RTPS only):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10317/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5846/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8402/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10224/)

Justification of Fast-RTPS only is that no issues came up in previous full CI run related to different RMW layers.

File: osrf_testing_tools_cpp/src/memory_tools/vendor/bombela/backward-cpp/BackwardConfig.cmake [hunk 1]
```diff
@@ -50,7 +50,7 @@ endif()
 ###############################################################################
 # CONFIGS
 ###############################################################################
-if (${STACK_DETAILS_AUTO_DETECT})
+if (${STACK_DETAILS_AUTO_DETECT} AND NOT WIN32)
```
Comment (jacobperron) [inline osrf_testing_tools_cpp/src/memory_tools/vendor/bombela/backward-cpp/BackwardConfig.cmake:53]: Should we document this change somehow for posterity? I.e. if we update Backward in a similar way in the future, is it possible we will lose this change accidentally?

Comment (wjwwood) [reply osrf_testing_tools_cpp/src/memory_tools/vendor/bombela/backward-cpp/BackwardConfig.cmake:53]: Yes, but I don't know where is the best place to do that. You will see the commit on top when you try to do a subtree update. I planned on opening a pull request to them, but I'm trying to prioritize other stuff atm.

Any idea on where to add a note that wouldn't get lost?

Comment (jacobperron) [reply osrf_testing_tools_cpp/src/memory_tools/vendor/bombela/backward-cpp/BackwardConfig.cmake:53]: I don't know either.

> You will see the commit on top when you try to do a subtree update.

That might be good enough.

**Comment by wjwwood** [comment]
All of the test failures are on nightlys and I think are flaky. The remaining CMake warnings are to do with ASSIMP, and the remaining MSBuild warnings are TF related.

**Pull request merged**